x/supply/keeper/keeper.go

```diff
@@ -92,7 +91,12 @@
 
 // GetTotalSupply returns the current total supply in the system
 func (k Keeper) GetTotalSupply(ctx sdk.Context) sdk.Coins {
-	return sdk.NewCoins(k.bankKeeper.GetSupply(ctx, config.BaseCoinUnit))
+	var totalSupply sdk.Coins
+	k.bankKeeper.IterateTotalSupply(ctx, func(coin sdk.Coin) bool {
+		totalSupply = totalSupply.Add(coin)
+		return false
+	})
+	return totalSupply
```

Comment:
Doesn't return all total supplies of all coin denoms now?

Reply:
yes? you mean only return BaseCoinUnit?

Reply:
It only returns basecro previously? Is there a bug inside the GetSupply method?

Reply:
GetSupply is for single denom, need switch to IterateTotalSupply if we need all denoms

Reply:
is the negative amount a bug in the GetSupply method?
Is this PR a bug fix, or just change it from returning single denom to multiple ones?

Reply:
yes, it's bug in query supply liquid, the fix is change to return for all denoms

Reply:
i see, because the other operands in the calculation contains all denoms?

Reply:
yea, when moduleAccount contains non BaseCoinUnit